independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Injection.java

```diff
@@ -320,7 +320,13 @@ private static List<AnnotationInstance> getAllInjectionPoints(BeanDeployment bea
             }
             AnnotationInstance inject = beanDeployment.getAnnotation(method, name);
             if (inject != null) {
-                injectAnnotations.add(inject);
+                if (Modifier.isStatic(method.flags())) {
+                    LOGGER.warn("An initializer method must be non-static - ignoring: "
```

Comment:
As this is a user error, I'd rather just throw an exception.

Reply:
Makes sense. I wonder why the spec does not treat a static initializer as a definition error :thinking:.

Reply:
See a few lines above -- this is how we treat `static` fields, I just follow suit :-)

Reply:
I know but it's never late to change your mind! ;-)

Reply:
It is. If we thrown an exception, ArC wouldn't even be able to process the AtInject TCK classes :-)

Reply:
> See a few lines above -- this is how we treat static fields, I just follow suit :-)

Yes, I noticed that and thought we could change both.

> It is. If we thrown an exception, ArC wouldn't even be able to process the AtInject TCK classes :-)

Hm, I see. In that case it'd be either a material for the strictness switch or we'd need to settle just for warning.

Reply:
The AtInject specification allows optional `static` injection, and the AtInject TCK has a switch to test this. We [will] set that switch to `false`, obviously, but we still need to be able to process those classes.

Using the strictness switch here would be an option, though not exactly intuitive (I'd expect code that works with strictness enabled to also work without strictness enabled).

Reply:
Not, I think it simply had conditional tests that verify the same functionality even for static variants if enabled. CDI has this disabled.
See https://github.com/eclipse-ee4j/injection-tck/blob/master/src/main/java/org/atinject/tck/Tck.java#L117-L119
And the related class - https://github.com/eclipse-ee4j/injection-tck/blob/master/src/main/java/org/atinject/tck/auto/Convertible.java#L474

Reply:
Ah, sorry for double answer, didn't see Ladislav's comment until I posted mine.

@Ladicek I agree that in this case it's better to keep the warning. However, a comment in the code (near that warning perhaps?) would be really nice :)

Reply:
Good point. I'll add a comment there.

Reply:
Done.